docs/proposals/odo-v3-cli/odo-init.md

```diff
@@ -0,0 +1,94 @@
+
+# `odo init`
+
+[#5297](https://github.com/redhat-developer/odo/issues/5297)
+
+The result of running `odo init` command should be local devfile.yaml saved in the current directory, and starter project extracted in the current directory (if user picked one)
+
+Running `odo init` in non-empty directory exits with error
+
+```
+The current directory is not empty. You can bootstrap new component only in empty directory.
+If you have existing code that you want to deploy use `odo deploy` or use `odo dev` command to quickly iterate on your component.
+```
+
+Command should use registries as configured in `odo registry` command. If there is multiple registries configured it should use all of them.
+
+## Flags
+
+- `--name` (string) - name of the component (required)
+- `--devfile` (string) - name of the devfile in devfile registry (required if `--devfile-path` is not defined)
+- `--registry` - (string) name of the devfile registry (as configured in `odo registry`). It can be used in combination with `--devfile`, but not with `--devfile-path` (optional)
+- `--starter`  (string) - name of the  starter project (optional)
+- `--devfile-path` (string) - path to a devfile. This is alternative to using devfile from Devfile registry. It can be local filesystem path or http(s) URL (required if `--devfile` is not defined)
+- `-o` (string) output information in a specified format (json).
```

Comment:
What is the expected information in output? The devfile.yaml file in json format?